src/Relude/Debug.hs

```diff
@@ -120,6 +121,18 @@ traceShowId :: Show a => a -> a
 traceShowId = Debug.traceShowId
 {-# WARNING traceShowId "'traceShowId' remains in code" #-}
 
+{- |
+Like 'traceShowId', but uses a provided function to convert
+the argument to a value with the 'Show' constraint.
+
+>>> fst $ traceShowWith fst (1, id)
+1
+1
```

Comment:
Nice example! But I think that it would make the meaning of the `traceShowWith` function even clearer if we won't use `fst` on the result

```suggestion
>>> traceShowWith fst (1, "ABC")
1
(1,"ABC")
```

Reply:
Ah, but with this I wanted to explicitly point out that you can use this function on things which do not have `Show` instance.

Reply:
Added both examples.